app/javascript/components/forms/SigninForm.jsx

```diff
@@ -45,7 +45,8 @@ export default function SigninForm() {
       <Row>
         <Col>
           <BootstrapForm.Group className="mb-2" controlId="formBasicCheckbox">
-            <BootstrapForm.Check type="checkbox" label="Remember me" className="small" />
+            {/* eslint-disable-next-line react/jsx-props-no-spreading */}
+            <BootstrapForm.Check type="checkbox" label="Remember me" className="small" {...methods.register('extend_session')} />
```

Comment:
I'm curious how the `{...methods.register('extend_session')}` works?

Reply:
It's from React Hook Form https://react-hook-form.com/api/useform/register

It's basically the thing that let's you build the object that is being sent in the request.

Ex:
register("firstName") --> {firstName: 'value'}

Reply:
> It's from React Hook Form https://react-hook-form.com/api/useform/register
> 
> It's basically the method that builds the object that is being sent in the request.
> 
> Ex: register("firstName") --> {firstName: 'value'} register("name.firstName") --> {name: { firstName: 'value' }}

Exactly!
Good explanation @scouillard 

It achieves that by returning a prop object defined according to `react-hook-form` config made when invoking its hook `useForm`.
The register props registers any kind of HTML inputs or select elements to `react-hook-form` so that it can control and keep hold of it.
It returns the fallowing props -> `name:, ref:, onChange, onBlur`.

Internally the registration process is a mapping of the registration name to a ref object.
More on that [HERE](https://react-hook-form.com/api/useform/register).

Reply:
Thanks for the explanation guys :+1: